src/components/monster.ts

```diff
@@ -64,6 +62,32 @@ export class Monster extends EventManager {
     // });
   }
 
+  checkHitboxDistance(event) {
+    /*
+    Note: Orginally used to check if within mouth distance. But inaccurate with the rive monster.
+    Might be re-use again in the future.
+    // const distance = Math.sqrt(
+    //   (cursorX - this.x - this.width / 3.5) ** 2 +
+    //   (cursorY - this.y - (this.width / 2) / 1.8)** 2 //Adjusted the divisor to lower the target point.
+    // );
+    */
+    const rect = this.canvasElement.getBoundingClientRect();
+    const x = event.clientX - rect.left;
+    const y = event.clientY - rect.top;
+
+    //Adjust this range factor to control how big is the hit box for dropping stones.
+    const rangeFactorX = 70; //SUBCTRACT FROM CENTER TO LEFT, ADD FROM CENTER TO RIGHT.
+    const rangeFactoryY = 50; //SUBCTRACT FROM CENTER TO TOP, ADD FROM CENTER TO BOTTOM.
+
+    const monsterXCenter = this.game.width / 2;
+    //Note: Rive height is currently always half of width. This might change when new rive files are to be implemented/
+    const monsterYCenter = monsterXCenter / 2; //Create different sets of height for multiple rive files or adjust this for height when replacing the current rive monster.
+    const isWithinHitboxY = y <= (monsterYCenter + rangeFactoryY) && y >= (monsterYCenter - rangeFactoryY);
+    const isWithinHitboxX = x <= (monsterXCenter + rangeFactorX) && x >= (monsterXCenter - rangeFactorX);
```

Comment:
As a perf improvement,

`(monsterYCenter + rangeFactoryY) && y >= (monsterYCenter - rangeFactoryY);` and  `(monsterXCenter + rangeFactorX) && x >= (monsterXCenter - rangeFactorX);` can be calculated on instantiation.

The app has a reinitialize logic when screen resizes so we only need to determine those once and just compare x and y respectively.